projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php

```diff
@@ -43,6 +78,10 @@ public function init_hooks() {
 			add_action( 'wp_footer', array( $this, 'print_instant_search_sidebar' ) );
 		} else {
 			add_action( 'update_option', array( $this, 'track_widget_updates' ), 10, 3 );
+			// The priority has to be lower than 10 to run before _wp_sidebars_changed.
+			// Which migrates widgets from old theme to the new one.
+			add_action( 'after_switch_theme', array( $this, 'save_old_sidebars_widgets' ), 5 );
```

Comment:
I'm guessing that `pre_update_option_sidebars_widgets` is fired after `after_switch_theme`. Is this correct?

Reply:
Yep. The process detected theme change is hooked to `init` by default, it fires `after_switch_theme` when it detects theme change and it'll update theme options afterwards, which would trigger `pre_update_option_sidebars_widgets`.

If it doesn't detect theme change, then the whole thing wouldn't run at all.